From 749ec77870dd3972eefa448a3c33715c53644e65 Mon Sep 17 00:00:00 2001
From: Farzin Kazemzadeh <itisFarzin@proton.me>
Date: Sun, 7 Dec 2025 07:39:07 +0000
Subject: [PATCH 49/50] Revert "Fix brightness slider curve for some devices"

* Thanks ChonDoe @ Telegram for pointing out the brightness issue

This reverts commit bb75dfe212b6daf031224d0c5150fabae14b8672.
---
 .../settingslib/display/BrightnessUtils.java       | 14 +++++---------
 .../brightness/shared/model/GammaBrightness.kt     | 14 +++++++-------
 2 files changed, 12 insertions(+), 16 deletions(-)

diff --git a/packages/SettingsLib/DisplayUtils/src/com/android/settingslib/display/BrightnessUtils.java b/packages/SettingsLib/DisplayUtils/src/com/android/settingslib/display/BrightnessUtils.java
index ceecd76ddd81..3c06f8ae9972 100644
--- a/packages/SettingsLib/DisplayUtils/src/com/android/settingslib/display/BrightnessUtils.java
+++ b/packages/SettingsLib/DisplayUtils/src/com/android/settingslib/display/BrightnessUtils.java
@@ -19,18 +19,14 @@ package com.android.settingslib.display;
 import android.os.SystemProperties;
 import android.util.MathUtils;
 
-import com.android.internal.display.BrightnessSynchronizer;
-
 /** Utility methods for calculating the display brightness. */
 public class BrightnessUtils {
 
     public static final boolean sDisableGammaConversion =
             SystemProperties.getBoolean("sys.brightness.disable_gamma_conversion", false);
 
-    public static final boolean LowGammaBrightness = Boolean.parseBoolean(SystemProperties.get("persist.sys.phh.low_gamma_brightness", "false"));
-
     public static final int GAMMA_SPACE_MIN = 0;
-    public static final int GAMMA_SPACE_MAX = LowGammaBrightness ? 255 : 65535;
+    public static final int GAMMA_SPACE_MAX = 65535;
 
     // Hybrid Log Gamma constant values
     private static final float R = 0.5f;
@@ -106,8 +102,9 @@ public class BrightnessUtils {
         // it shouldn't be out of bounds.
         final float normalizedRet = MathUtils.constrain(ret, 0, 12);
 
-        return LowGammaBrightness ? MathUtils.constrain(BrightnessSynchronizer.brightnessIntToFloat(val),
-                         min, max) : MathUtils.lerp(min, max, normalizedRet / 12);
+        // Re-normalize to the range [0, 1]
+        // in order to derive the correct setting value.
+        return MathUtils.lerp(min, max, normalizedRet / 12);
     }
 
     /**
@@ -159,7 +156,6 @@ public class BrightnessUtils {
             ret = A * MathUtils.log(normalizedVal - B) + C;
         }
 
-        return LowGammaBrightness ? BrightnessSynchronizer.brightnessFloatToInt(
-                       MathUtils.constrain(val, min, max)) : Math.round(MathUtils.lerp(GAMMA_SPACE_MIN, GAMMA_SPACE_MAX, ret));
+        return Math.round(MathUtils.lerp(GAMMA_SPACE_MIN, GAMMA_SPACE_MAX, ret));
     }
 }
diff --git a/packages/SystemUI/src/com/android/systemui/brightness/shared/model/GammaBrightness.kt b/packages/SystemUI/src/com/android/systemui/brightness/shared/model/GammaBrightness.kt
index b7a9d5f535b4..7eba6268869c 100644
--- a/packages/SystemUI/src/com/android/systemui/brightness/shared/model/GammaBrightness.kt
+++ b/packages/SystemUI/src/com/android/systemui/brightness/shared/model/GammaBrightness.kt
@@ -23,13 +23,13 @@ import com.android.systemui.util.kotlin.pairwiseBy
 import kotlinx.coroutines.flow.Flow
 
 @JvmInline
-value class GammaBrightness(val value: Int) {
-    init {
-        require(value in BrightnessUtils.GAMMA_SPACE_MIN..BrightnessUtils.GAMMA_SPACE_MAX) {
-            "Value must be in range ${BrightnessUtils.GAMMA_SPACE_MIN} to ${BrightnessUtils.GAMMA_SPACE_MAX}"
-        }
-    }
-}
+value class GammaBrightness(
+    @IntRange(
+        from = BrightnessUtils.GAMMA_SPACE_MIN.toLong(),
+        to = BrightnessUtils.GAMMA_SPACE_MAX.toLong()
+    )
+    val value: Int
+)
 
 internal fun Flow<GammaBrightness>.logDiffForTable(
     tableLogBuffer: TableLogBuffer,
-- 
2.43.0

